src/CSharpIsNullAnalyzer/CSIsNull001.cs

```diff
@@ -44,6 +44,12 @@ public override void Initialize(AnalysisContext context)
             context.EnableConcurrentExecution();
             context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.ReportDiagnostics);
 
+            context.RegisterOperationAction(
+                ctxt =>
+                {
+                    // ctxt.Operation.
```

Comment:
@sharwell do you have any suggestions for detecting whether an expression is part of an expression tree? I imagine we need more than just the syntax tree, so I've looked at using IOperation, but it's not clear to me how to use that API either.

Reply:
https://github.com/dotnet/roslyn-analyzers/blob/762b08948cdcc1d94352fba681296be7bf474dd7/src/Utilities/Compiler/Extensions/IOperationExtensions.cs#L519-L522